charts/newrelic-infrastructure-v3/templates/NOTES.txt

```diff
@@ -13,6 +13,149 @@ Those components are needed to have the full experience on NROne kubernetes expl
 
 {{- end }}
 
+{{- if .Values.nodeSelector }}
+
+###########
+# WARNING #
+###########
+
+You have specified "nodeSelector" into values, however the new version of the chart
+support different nodeSelectors per each component, therefore the value specified has been ignored.
+
+Please use ksm.nodeSelector, controlPlane.nodeSelector and kubelet.nodeSelector.
+
+{{- end }}
+
+{{- if .Values.nodeAffinity }}
+
+###########
+# WARNING #
+###########
+
+You have specified "nodeAffinity" into values, however the new version of the chart
+support different nodeAffinity options per component, therefore the value specified has been ignored.
+
+Please use ksm.affinity.nodeAffinity, controlPlane.affinity.nodeAffinity and kubelet.affinity.nodeAffinity.
+
+{{- end }}
```

Comment:
Same as above

Reply:
nodeAffinity are already present in default in case of control plane. We can give precedence to legacy value.
I am not totally sure we should give precedence to an "old" config that was thought for a different object..
WDYT?

Reply:
🤔 Fair point, I don't like either option:
- If we ignore the old value, we will be deploying kubelet in pods that were not meant to be monitored
- If we honor the old value, we can also deploy the CP daemonset in a bunch of worker nodes

What about applying the legacy affinity/selector to kubelet/ksm deployments only? Keep the default one for the control plane DS (avoid overriding it) and emit a warning.
It increases complexity slightly but I think it could be a good approach.

Reply:
did that